grpc-interface-support/src/cpp.py

```diff
@@ -18,19 +18,323 @@
+
+        package_pieces = self.__proto_file.get_package().split(".")
+
+        source_content: List[str] = capture_textfile_area(
+            open(grpc_source_path, encoding="utf-8"),
+            f"{service_name}::Service::~Service() {{",
+            f"}}  // namespace {package_pieces[0]}",
+        )
+
+        # skin initial 2 lines b/c they are always empty
+        return source_content[2 : len(source_content) - 2]
+
+
+class CppGrpcServiceSdkGenerator(GrpcServiceSdkGenerator):  # type: ignore
+    def __init__(
+        self,
+        package_directory_path: str,
+        proto_file_handle: ProtoFileHandle,
+        verbose: bool,
+    ):
+        self.__package_directory_path = package_directory_path
+        self.__proto_file_handle = proto_file_handle
+        self.__verbose = verbose
+
+    def __get_binary_path(self, binary_name: str) -> str:
+        path = shutil.which(binary_name)
+        if path is None:
+            raise KeyError(f"{binary_name!r} missing!")
+        return path
+
+    def __invoke_code_generator(self, include_path: str, output_path: str) -> None:
+        print("Invoking gRPC code generator")
+        args = [
+            self.__get_binary_path("protoc"),
+            f"--plugin=protoc-gen-grpc={self.__get_binary_path('grpc_cpp_plugin')}",
+            f"-I{include_path}",
+            f"--cpp_out={output_path}",
+            f"--grpc_out={output_path}",
+            self.__proto_file_handle.file_path,
+        ]
+        subprocess.check_call(
+            args,
+            cwd=include_path,
+            env=os.environ,
+            stdout=subprocess.DEVNULL if not self.__verbose else None,
+        )
+
+    def generate_package(self, client_required: bool, server_required: bool) -> None:
+        self.__invoke_code_generator(
+            str(Path(self.__proto_file_handle.file_path).parent),
+            self.__package_directory_path,
+        )
+
+        service_name = self.__proto_file_handle.get_service_name()
+
+        files_to_copy: List[templates.CopySpec] = []
+
+        if client_required:
+            files_to_copy.extend(self.__get_service_client_files(service_name))
+
+        if server_required:
+            files_to_copy.extend(self.__get_service_server_files(service_name))
+
+        templates.copy_templates(
+            get_template_dir(),
+            self.__package_directory_path,
+            files_to_copy,
+            self.__get_template_variables(),
+        )
+
+    def __get_template_variables(self) -> Dict[str, str]:
+        service_name = self.__proto_file_handle.get_service_name()
+
+        return {
+            "service_name": service_name,
+            "service_name_lower": service_name.lower(),
+            "service_name_camel_case": to_camel_case(service_name),
+            "package_id": self.__proto_file_handle.get_package().replace(".", "::"),
+            "core_sdk_version": str(conan_helper.get_required_sdk_version()),
+        }
```

Comment:
since you already defined that here, please make use of it in the entire file

Reply:
Done